DllImportGenerator/TestAssets/Benchmarking/CodeGenMonitor.cs

```diff
@@ -0,0 +1,184 @@
+            {
+                Title = "Compare generated code",
+                Pane1Title = "Code Gen 1",
+                Pane2Title = "Code Gen 2",
+                SelectionTitle = "All methods",
+            };
+        }
+
+        /// <summary>
+        /// Get the generated code for the supplied method.
+        /// </summary>
+        /// <param name="fullyQualifiedMethodName">Fully qualified method name</param>
+        /// <param name="metadataToken">[Optional] method's metadata token - used to clarify overloads.</param>
+        /// <param name="retryCount">Number of times to retry if method isn't immediately found</param>
+        /// <returns>A <see cref="MethodCodeGen"/> instance</returns>
+        public MethodCodeGen GetLastCodeGenFor(string fullyQualifiedMethodName, int metadataToken = MetadataTokenNil, int retryCount = 5)
+        {
+            return GetLastCodeGenForAsync(fullyQualifiedMethodName, metadataToken, retryCount).Result;
+        }
+
+        /// <summary>
+        /// Asynchronously get the generated code for the supplied method.
+        /// </summary>
+        /// <param name="fullyQualifiedMethodName">Fully qualified method name</param>
+        /// <param name="metadataToken">[Optional] method's metadata token - used to clarify overloads.</param>
+        /// <param name="retryCount">Number of times to retry if method isn't immediately found</param>
+        /// <returns>A <see cref="MethodCodeGen"/> instance</returns>
+        public async Task<MethodCodeGen> GetLastCodeGenForAsync(string fullyQualifiedMethodName, int metadataToken = MetadataTokenNil, int retryCount = 5)
+        {
+            for (int i = 0; i < retryCount; ++i)
+            {
+                lock (this.methods)
+                {
+                    if (this.methods.TryGetValue(fullyQualifiedMethodName, out List<MethodCodeGen> mcgs))
+                    {
+                        var ret = mcgs.Last();
+                        if (metadataToken != MetadataTokenNil)
+                        {
+                            // Return the last matching element in the collection
+                            var lastMaybe = mcgs.FindLast((m) => m.MetadataToken == (uint)metadataToken);
+                            if (lastMaybe is not null)
+                            {
+                                ret = lastMaybe;
+                            }
+                        }
+
+                        return ret;
+                    }
+                }
+
+                await Task.Delay(this.retryDelayInMs);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Generate a static HTML page to compare all collected data.
+        /// </summary>
+        /// <param name="pattern">Regular expression to filter through all collected data.</param>
+        /// <param name="waitForMonitorInMs">Time delay to wait prior to accessing current state</param>
```

Comment:
`waitForMonitorInMs` seems like an odd thing to bake the API to me?

Reply:
Since the system is backed by ETW and the usage pattern is call this at the end it is best to wait a second or two. I will leave this for now.